File: homework_01/task08.py
#task 08
import math
def solv_quadratic_equation(a, b, c):
    if not a: return "Enter valid constans"
    else:
        d = pow(b, 2) - 4*a*c
        if d == 0:
            x = -b / 2 * a
            return "Solution is {}".format(x)
        elif d < 0:
            return "Solution does not exists"
        x1 = ((-b - math.sqrt(d)) / (2 * a))
        x2 = ((-b + math.sqrt(d)) / (2 * a))
        return "Solution are {0} and {1}".format(x1, x2)
# test
print(solv_quadratic_equation(1, 2, 1))
print(solv_quadratic_equation(0, 4, -5))
print(solv_quadratic_equation(3, -8, 12))
print(solv_quadratic_equation(5, -13, 6))

File: Homework/homework_01/task08.js
"use strict";

const findRoots = (a, b, c) => {
    if (a === 0) {
        return "Enter valid constans";
    }
    else {
        let d = Math.pow(b, 2) - 4 * a * c;
        // console.log(d);

        if (d < 0) {
            return "Solution does not exists";
        }
        else if (d === 0) {
            let x = -b / 2 * a;
            return `Solution is ${x}`;
        } else {
            let x1 = ((-b - Math.sqrt(d)) / (2 * a));
            let x2 = ((-b + Math.sqrt(d)) / (2 * a));
            return `Solution are ${x1} and ${x2}`;
        }
    }
}
let test = [[1, 2, 1], [0, 4, -5], [3, -8, 12], [5, -13, 6]];
for (let i = 0; i < test.length; i++) {
    let a = test[i][0];
    let b = test[i][1];
    let c = test[i][2];
    console.log(findRoots(a, b, c));
}

